### PR TITLE
Created activation functions in Pytorch

### LINKED ISSUE #5231: [Term Entry] PyTorch nn Activation Functions
### Reason/inspiration (optional)

We would like a new entry on the `Activation Functions` in PyTorch. The entry should go in a new file under `content/pytorch/concepts/nn/terms/activation-functions/activation-functions.md`.      _x000D_
  _x000D_
The entry should include:  _x000D_
  _x000D_
- An introduction to the concept     _x000D_
  _x000D_
- A `Syntax` section that provides the syntax for the concept (if applicable)     _x000D_
  _x000D_
- An `Example` section that provides an example demonstrating the concept in use (if applicable)     _x000D_
  _x000D_
Please refer to the [concept entry template](https://github.com/Codecademy/docs/blob/main/documentation/concept-entry-template.md), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [ ] Issue is assigned

### PR DESCRIPTION
**Pull request by arisdelacruz**
### Description

This is an attempt to create an entry for the Activation functions in Pytorch.

### Issue Solved

Closes #5231 

### Type of Change

- Adding a new entry

### Checklist

- [/] All writings are my own.
- [/] My entry follows the Codecademy Docs style guide.
- [/] My changes generate no new warnings.
- [/] I have performed a self-review of my own writing and code.
- [/] I have checked my entry and corrected any misspellings.
- [/] I have made corresponding changes to the documentation if needed.
- [/] I have confirmed my changes are not being pushed from my forked `main` branch.
- [/] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [/] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

**Comment by mamtawardhani** [comment]
Hey @arisdelacruz the path of the file seems to be incorrect. It should be `content/pytorch/concepts/nn/activation-functions/activation-functions.md`, but your entry is missing the `terms` folder. Please make that change

File: content/pytorch/concepts/nn/activation-functions/activation-functions.md [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+    x = self.fc2(x)
+    return x
+
+# Create a model instance
+model = SimpleNN()
+```
+
+## ReLU6
+
+ReLU6 is a variation of the ReLU activation function that returns the input if it is positive and less than 6, and zero otherwise. It is defined as:
```
Comment (mamtawardhani) [inline content/pytorch/concepts/nn/activation-functions/activation-functions.md:52]:
```suggestion
ReLU6 is a variation of the ReLU activation function that returns the input if it is positive and less than or equal to 6, and returns 0 if the input is negative or greater than 6. It is mathematically defined as:
```

Comment (mamtawardhani) [reply content/pytorch/concepts/nn/activation-functions/activation-functions.md:52]: If I'm not wrong, it should be "less than and equal to 6", do recheck once though

File: content/pytorch/concepts/nn/activation-functions/activation-functions.md [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+
+## Leaky ReLU
+
+Leaky ReLU is a variation of the ReLU activation function that returns the input if it is positive, and a small fraction of the input otherwise. It is defined as:
+
+```pseudo
+f(x) = max(0.01x, x)
+```
+
+Leaky ReLU helps prevent the dying ReLU problem, where neurons can become inactive and stop learning. By allowing a small gradient for negative inputs, Leaky ReLU ensures that all neurons in the network continue to learn.
```
Comment (mamtawardhani) [inline content/pytorch/concepts/nn/activation-functions/activation-functions.md:88]:
```suggestion
Leaky ReLU helps prevent the dying ReLU problem, where neurons can become inactive and stop learning. By allowing a small gradient for negative inputs, Leaky ReLU ensures that all neurons in the network continue to learn, promoting better training and performance.
```

File: content/pytorch/concepts/nn/activation-functions/activation-functions.md [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+    x = self.fc2(x)
+    return x
+
+# Create a model instance
+model = SimpleNN()
+```
+
+## Sigmoid
+
+Sigmoid is another popular activation function that returns a value between 0 and 1. It is defined as:
```
Comment (mamtawardhani) [inline content/pytorch/concepts/nn/activation-functions/activation-functions.md:112]:
```suggestion
Sigmoid is a popular activation function that returns a value between 0 and 1. It is mathematically defined as:
```

File: content/pytorch/concepts/nn/activation-functions/activation-functions.md [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+
+```pseudo
+f(x) = 1 / (1 + e^(-x))
+```
+
+Sigmoid is commonly used in binary classification problems where the output needs to be between 0 and 1. It is also used in the output layer of neural networks to predict probabilities.
```
Comment (mamtawardhani) [inline content/pytorch/concepts/nn/activation-functions/activation-functions.md:118]:
```suggestion
The Sigmoid function is commonly used in binary classification problems where the output needs to be constrained between 0 and 1. It is also frequently utilized in the output layer of neural networks to predict probabilities, making it suitable for tasks that require a binary outcome.
```

File: content/pytorch/concepts/nn/activation-functions/activation-functions.md [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+    x = self.fc2(x)
+    return x
+
+# Create a model instance
+model = SimpleNN()
+```
+
+## Tanh
+
+Tanh is another activation function that returns a value between -1 and 1. It is defined as:
```
Comment (mamtawardhani) [inline content/pytorch/concepts/nn/activation-functions/activation-functions.md:142]:
```suggestion
Tanh (hyperbolic tangent) is an activation function that returns a value between -1 and 1. It is mathematically defined as:
```

File: content/pytorch/concepts/nn/activation-functions/activation-functions.md [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+
+```pseudo
+f(x) = (e^x - e^(-x)) / (e^x + e^(-x))
+```
+
+Tanh is similar to the Sigmoid function but returns values between -1 and 1. It is commonly used in the hidden layers of neural networks to introduce non-linearity.
```
Comment (mamtawardhani) [inline content/pytorch/concepts/nn/activation-functions/activation-functions.md:148]:
```suggestion
Tanh is similar to the Sigmoid function but differs in that it outputs values within the range of -1 to 1. This property makes it particularly useful in the hidden layers of neural networks, as it helps introduce non-linearity and can lead to improved convergence during training.
```

File: content/pytorch/concepts/nn/activation-functions/activation-functions.md [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+    x = self.fc2(x)
+    return x
+
+# Create a model instance
+model = SimpleNN()
+```
+
+## Softmax
+
+Softmax is an activation function that returns a probability distribution over multiple classes. It is defined as:
```
Comment (mamtawardhani) [inline content/pytorch/concepts/nn/activation-functions/activation-functions.md:172]:
```suggestion
Softmax is an activation function that returns a probability distribution over multiple classes. It is mathematically defined as:
```

File: content/pytorch/concepts/nn/activation-functions/activation-functions.md [hunk 8]
```diff
@@ -0,0 +1,199 @@
...
+
+```pseudo
+f(x_i) = e^(x_i) / sum(e^(x_j)) for j = 1 to n
+```
+
+Softmax is commonly used in the output layer of neural networks for multi-class classification problems. It ensures that the output values sum to 1, making it easier to interpret the model's predictions as probabilities.
```
Comment (mamtawardhani) [inline content/pytorch/concepts/nn/activation-functions/activation-functions.md:178]:
```suggestion
Softmax is commonly used in the output layer of neural networks for multi-class classification problems. It ensures that the output values sum to 1, allowing the model's predictions to be interpreted as probabilities for each class. This property makes it particularly useful for tasks where multiple classes are present, providing a clear way to identify the predicted class.
```

File: content/pytorch/concepts/nn/activation-functions/activation-functions.md [hunk 9]
```diff
@@ -0,0 +1,199 @@
...
+  - 'py-torch-for-classification'
+---
+
+The **`nn Activation Functions`** in Pytorch are a set of pre-built activation functions that are essential for building neural networks. Activation functions are mathematical equations that determine the output of a neural network. The function is attached to each neuron in the network and determines whether it should be activated or not, based on whether each neuron's input is relevant for the model's prediction. Activation functions also help normalize the output of each neuron to a range between 1 and 0 or between -1 and 1.
+
+Activation functions are used to introduce non-linearity to the neural network. Common activation functions include `ReLU`, `ReLU6`, `Leaky ReLU`, `Sigmoid`, `Tanh`, and `Softmax`. Activation functions are applied to the output of each neuron in the network.
+
+## ReLU (Rectified Linear Unit)
+
+ReLU is a popular activation function that returns the input if it is positive, and zero otherwise. It is defined as:
```
Comment (mamtawardhani) [inline content/pytorch/concepts/nn/activation-functions/activation-functions.md:22]:
```suggestion
ReLU is a popular activation function that returns the input if it is positive, and zero otherwise. It is mathematically defined as:
```

File: content/pytorch/concepts/nn/activation-functions/activation-functions.md [hunk 10]
```diff
@@ -0,0 +1,199 @@
...
+    x = self.fc2(x)
+    return x
+
+# Create a model instance
+model = SimpleNN()
+```
+
+## Leaky ReLU
+
+Leaky ReLU is a variation of the ReLU activation function that returns the input if it is positive, and a small fraction of the input otherwise. It is defined as:
```
Comment (mamtawardhani) [inline content/pytorch/concepts/nn/activation-functions/activation-functions.md:82]:
```suggestion
Leaky ReLU is a variation of the ReLU activation function that returns the input if it is positive, and a small fraction of the input otherwise. It is mathematically defined as:
```

**Review by mamtawardhani — CHANGES_REQUESTED** [review]
Hey @arisdelacruz thank you for contributing to Codecademy Docs, the entry is nicely written! 😄 

I've suggested a few modifications, could review and make the changes at your earliest convenience? Also, the path of the file seems to be incorrect, the `terms` folder is missing, please correct that as well.
Thank you! 😄

**Comment by arisdelacruz** [comment]
> Hey @arisdelacruz the path of the file seems to be incorrect. It should be `content/pytorch/concepts/nn/activation-functions/activation-functions.md`, but your entry is missing the `terms` folder. Please make that change

@mamtawardhani 

Sorry, shall I do a pull and push for this one? Thanks for the guidance.

**Comment by mamtawardhani** [comment]
> > Hey @arisdelacruz the path of the file seems to be incorrect. It should be `content/pytorch/concepts/nn/activation-functions/activation-functions.md`, but your entry is missing the `terms` folder. Please make that change
> 
> @mamtawardhani
> 
> Sorry, shall I do a pull and push for this one? Thanks for the guidance.

nope, can be done here itself. Just take this file and folder,and put it inside the terms folder and commit and push the changes again

**Comment by arisdelacruz** [comment]
@mam

> > > Hey @arisdelacruz the path of the file seems to be incorrect. It should be `content/pytorch/concepts/nn/activation-functions/activation-functions.md`, but your entry is missing the `terms` folder. Please make that change
> > 
> > 
> > @mamtawardhani
> > Sorry, shall I do a pull and push for this one? Thanks for the guidance.
> 
> nope, can be done here itself. Just take this file and folder,and put it inside the terms folder and commit and push the changes again

@mamtawardhani 

Am I on the right track now? Thanks

**Review by mamtawardhani — APPROVED** [review]
Thank you for contributing to Codecademy Docs @arisdelacruz 😄 

The entry looks good for a second rounf of review! 🚀

File: content/pytorch/concepts/nn/terms/activation-functions/activation-functions.md [hunk 11]
```diff
@@ -0,0 +1,199 @@
...
+
+## ReLU6
+
+ReLU6 is a variation of the ReLU activation function that returns the input if it is positive and less than or equal to 6, and returns 0 if the input is negative or greater than 6. It is mathematically defined as:
+
+```pseudo
+f(x) = min(max(0, x), 6)
+```
+
+ReLU6 is useful when you want to limit the output of the activation function to a specific range, such as between 0 and 6.
```
Comment (Radhika-okhade) [inline content/pytorch/concepts/nn/terms/activation-functions/activation-functions.md:58]:
```suggestion
ReLU6 is useful when you want to limit the output of the activation function to a specific range, such as between `0` and `6`.

```

File: content/pytorch/concepts/nn/terms/activation-functions/activation-functions.md [hunk 12]
```diff
@@ -0,0 +1,199 @@
...
+    x = self.fc2(x)
+    return x
+
+# Create a model instance
+model = SimpleNN()
+```
+
+## Sigmoid
+
+Sigmoid is a popular activation function that returns a value between 0 and 1. It is mathematically defined as:
```
Comment (Radhika-okhade) [inline content/pytorch/concepts/nn/terms/activation-functions/activation-functions.md:112]:
```suggestion
Sigmoid is a popular activation function that returns a value between `0` and `1`. It is mathematically defined as:

```

File: content/pytorch/concepts/nn/terms/activation-functions/activation-functions.md [hunk 13]
```diff
@@ -0,0 +1,199 @@
...
+
+```pseudo
+f(x) = 1 / (1 + e^(-x))
+```
+
+The Sigmoid function is commonly used in binary classification problems where the output needs to be constrained between 0 and 1. It is also frequently utilized in the output layer of neural networks to predict probabilities, making it suitable for tasks that require a binary outcome.
```
Comment (Radhika-okhade) [inline content/pytorch/concepts/nn/terms/activation-functions/activation-functions.md:118]:
```suggestion
The Sigmoid function is commonly used in binary classification problems where the output needs to be constrained between `0` and `1`. It is also frequently utilized in the output layer of neural networks to predict probabilities, making it suitable for tasks that require a binary outcome.

```

File: content/pytorch/concepts/nn/terms/activation-functions/activation-functions.md [hunk 14]
```diff
@@ -0,0 +1,199 @@
...
+    x = self.fc2(x)
+    return x
+
+# Create a model instance
+model = SimpleNN()
+```
+
+## Tanh
+
+Tanh (hyperbolic tangent) is an activation function that returns a value between -1 and 1. It is mathematically defined as:
```
Comment (Radhika-okhade) [inline content/pytorch/concepts/nn/terms/activation-functions/activation-functions.md:142]:
```suggestion
Tanh (hyperbolic tangent) is an activation function that returns a value between `-1` and `1`. It is mathematically defined as:

```

File: content/pytorch/concepts/nn/terms/activation-functions/activation-functions.md [hunk 15]
```diff
@@ -0,0 +1,199 @@
...
+
+```pseudo
+f(x) = (e^x - e^(-x)) / (e^x + e^(-x))
+```
+
+Tanh is similar to the Sigmoid function but differs in that it outputs values within the range of -1 to 1. This property makes it particularly useful in the hidden layers of neural networks, as it helps introduce non-linearity and can lead to improved convergence during training.
```
Comment (Radhika-okhade) [inline content/pytorch/concepts/nn/terms/activation-functions/activation-functions.md:148]:
```suggestion
Tanh is similar to the Sigmoid function but differs in that it outputs values within the range of `-1` to `1`. This property makes it particularly useful in the hidden layers of neural networks, as it helps introduce non-linearity and can lead to improved convergence during training.

```

File: content/pytorch/concepts/nn/terms/activation-functions/activation-functions.md [hunk 16]
```diff
@@ -0,0 +1,199 @@
...
+
+## Softmax
+
+Softmax is an activation function that returns a probability distribution over multiple classes. It is mathematically defined as:
+
+```pseudo
+f(x_i) = e^(x_i) / sum(e^(x_j)) for j = 1 to n
+```
+
+Softmax is commonly used in the output layer of neural networks for multi-class classification problems. It ensures that the output values sum to 1, allowing the model's predictions to be interpreted as probabilities for each class. This property makes it particularly useful for tasks where multiple classes are present, providing a clear way to identify the predicted class.
```
Comment (Radhika-okhade) [inline content/pytorch/concepts/nn/terms/activation-functions/activation-functions.md:178]:
```suggestion
Softmax is commonly used in the output layer of neural networks for multi-class classification problems. It ensures that the output values sum to `1`, allowing the model's predictions to be interpreted as probabilities for each class. This property makes it particularly useful for tasks with multiple classes, providing a clear way to identify the predicted class.

```

**Review by Radhika-okhade — APPROVED** [review]

**Review by Radhika-okhade — APPROVED** [review]
Thank you for contributing to Codecademy docs. The entry looks good to be merged.
